api/src/main/java/io/opentelemetry/spi/TracerProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.spi;
+
+import io.opentelemetry.trace.Tracer;
+
+/**
+ * TracerProvider is a service provider for {@link Tracer}. Implementation should register
+ * implementation in <code>META-INF/services/io.opentelemetry.spi.TracerProvider</code>. <br>
+ * <br>
+ * A specific implementation can be specified by a system property {@code
+ * io.opentelemetry.spi.TracerProvider} with value of fully qualified class name.
+ */
+public interface TracerProvider {
```

Comment:
Why do we have this in spi rather than in the opentelemetry.trace?

Reply:
The SPI package makes it clear that this is an extension point. It's usually a good practice to put this into a separate package. 

We can move it to trace package if you prefer to do that.

Reply:
Should then move the OpenTelemetry class in the spi as well? Then everything about the implementation loading is in that package?

Reply:
I would prefer keeping this in the root package as it is the main access point.

I would like to move `loadTracer` method to `spi.TracerProvider` and change it to an abstract class.

Reply:
My 2 cents here is that the API was clear in terms of the exported packages. having a top level "spi" here does not make too much sense. I would like to consider a different structure where spi is the package that contains all related to service loading, or otherwise maybe move TracerProvider to tracer/spi?